### PR TITLE
Add Hub20 keyboard

### PR DESCRIPTION
**Pull request by joshajohnson**
## Description

Adds support for Hub20 numpad.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/hub20/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+       _______,  _______,  _______,    _______,  \
+       _______,  _______,  _______,    _______   \
+),
+    [3] = LAYOUT_all(  \
+                 _______,  _______,
+       _______,  _______,  _______,    _______,  \
+       _______,  _______,  _______,    _______,  \
+       _______,  _______,  _______,    _______,  \
+       _______,  _______,  _______,    _______,  \
+       _______,  _______,  _______,    _______   \
```
Comment (fauxpark) [inline keyboards/hub20/keymaps/default/keymap.c:51]:
```suggestion
    [0] = LAYOUT_all(
                 KC_MUTE,  KC_MPLY,
       KC_PMNS,  KC_PAST,  KC_PSLS,    MO_NLCK,
       KC_PPLS,    KC_P7,    KC_P8,      KC_P9,
       KC_PPLS,    KC_P4,    KC_P5,      KC_P6,
       KC_PENT,    KC_P1,    KC_P2,      KC_P3,
       KC_PENT,  KC_PDOT,    KC_P0,      KC_P0
),
    [1] = LAYOUT_all(
                   RESET,  _______,
       RGB_TOG, RGB_RMOD,  RGB_MOD,    _______,
       _______,  RGB_VAD,  RGB_VAI,    _______,
       _______,  RGB_HUD,  RGB_HUI,    _______,
       _______,  RGB_SAD,  RGB_SAI,    _______,
       _______,  _______,  _______,    _______
```

File: keyboards/hub20/keymaps/macro/keymap.c [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+       KC_M,  KC_N,  KC_O,  KC_P,  \
+       KC_Q,  KC_R,  KC_S,  KC_T   \
+),
+    [_CTRL] = LAYOUT_all(  \
+                   RESET,  _______,
+       RGB_TOG, RGB_RMOD,  RGB_MOD,   TD(BASE),  \
+       _______,  RGB_VAD,  RGB_VAI,    _______,  \
+       _______,  RGB_HUD,  RGB_HUI,    _______,  \
+       _______,  RGB_SAD,  RGB_SAI,    _______,  \
+       _______,  _______,  _______,    _______   \
```
Comment (fauxpark) [inline keyboards/hub20/keymaps/macro/keymap.c:56]:
```suggestion
    [_BASE] = LAYOUT_all(
              KC_W,  KC_Z,
       KC_A,  KC_B,  KC_C,  TD(CTRL),
       KC_E,  KC_F,  KC_G,  KC_H,
       KC_I,  KC_J,  KC_K,  KC_L,
       KC_M,  KC_N,  KC_O,  KC_P,
       KC_Q,  KC_R,  KC_S,  KC_T
),
    [_CTRL] = LAYOUT_all(
                   RESET,  _______,
       RGB_TOG, RGB_RMOD,  RGB_MOD,   TD(BASE),
       _______,  RGB_VAD,  RGB_VAI,    _______,
       _______,  RGB_HUD,  RGB_HUI,    _______,
       _______,  RGB_SAD,  RGB_SAI,    _______,
       _______,  _______,  _______,    _______
```

File: keyboards/hub20/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+       _______,  _______,  _______,    _______,  \
+       _______,  _______,  _______,    _______   \
+),
+    [3] = LAYOUT_all(  \
+                 _______,  _______,
+       _______,  _______,  _______,    _______,  \
+       _______,  _______,  _______,    _______,  \
+       _______,  _______,  _______,    _______,  \
+       _______,  _______,  _______,    _______,  \
+       _______,  _______,  _______,    _______   \
```
Comment (fauxpark) [inline keyboards/hub20/keymaps/via/keymap.c:51]:
```suggestion
    [0] = LAYOUT_all(
                 KC_MUTE,  KC_MPLY,
       KC_PMNS,  KC_PAST,  KC_PSLS,    MO_NLCK,
       KC_PPLS,    KC_P7,    KC_P8,      KC_P9,
       KC_PPLS,    KC_P4,    KC_P5,      KC_P6,
       KC_PENT,    KC_P1,    KC_P2,      KC_P3,
       KC_PENT,  KC_PDOT,    KC_P0,      KC_P0
),
    [1] = LAYOUT_all(
                   RESET,  _______,
       RGB_TOG,  RGB_RMOD, RGB_MOD,    _______,
       _______,  RGB_VAD,  RGB_VAI,    _______,
       _______,  RGB_HUD,  RGB_HUI,    _______,
       _______,  RGB_SAD,  RGB_SAI,    _______,
       _______,  _______,  _______,    _______
),
    [2] = LAYOUT_all(
                 _______,  _______,
       _______,  _______,  _______,    _______,
       _______,  _______,  _______,    _______,
       _______,  _______,  _______,    _______,
       _______,  _______,  _______,    _______,
       _______,  _______,  _______,    _______
),
    [3] = LAYOUT_all(
                 _______,  _______,
       _______,  _______,  _______,    _______,
       _______,  _______,  _______,    _______,
       _______,  _______,  _______,    _______,
       _______,  _______,  _______,    _______,
       _______,  _______,  _______,    _______
```

File: keyboards/hub20/readme.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+For more information regarding the keyboard, please visit the [Hub20 Website](https://www.joshajohnson.com/hub20) or [GitHub Repo](https://github.com/joshajohnson/Hub20).
+
+* Keyboard Maintainer: [Josh Johnson](https://github.com/joshajohnson)
+* Hardware Supported: Hub20 PCB (STM32F072)
+* Hardware Availability: [Josh Johnson](https://www.joshajohnson.com/hub20)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make hub20:default:flash
```
Comment (fauxpark) [inline keyboards/hub20/readme.md:13]:
```suggestion
    make hub20:default

Flashing example for this keyboard:

    make hub20:default:flash
```

File: keyboards/hub20/rules.mk [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+MOUSEKEY_ENABLE = yes		# Mouse keys
+EXTRAKEY_ENABLE = yes		# Audio control and System control
+CONSOLE_ENABLE = yes		# Console for debug
+COMMAND_ENABLE = yes    	# Commands for debug and configuration
+SLEEP_LED_ENABLE = no  		# Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	    	# USB Nkey Rollover
+CUSTOM_MATRIX = no 			# Custom matrix file
+ENCODER_ENABLE = yes		# Rotary Encoder support
+TAP_DANCE_ENABLE = no 		# Support for tap dancing
+RGBLIGHT_ENABLE = yes		# Underglow
```
Comment (fauxpark) [inline keyboards/hub20/rules.mk:14]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
```

File: keyboards/hub20/bootloader_defs.h [hunk 6]
```diff
@@ -0,0 +1,7 @@
+/* Address for jumping to bootloader on STM32 chips. */
+/* It is chip dependent, the correct number can be looked up here (page 175):
+ * http://www.st.com/web/en/resource/technical/document/application_note/CD00167594.pdf
+ * This also requires a patch to chibios:
+ *  <tmk_dir>/tmk_core/tool/chibios/ch-bootloader-jump.patch
+ */
+#define STM32_BOOTLOADER_ADDRESS 0x1FFFC800
```
Comment (fauxpark) [inline keyboards/hub20/bootloader_defs.h:7]: This file can be deleted.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by joshajohnson** [comment]
Thanks for the reviews @drashna and @fauxpark :)

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**